code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -48,21 +49,24 @@
 		balloon_alert(user, "You start cranking")
 		while(cell.charge < gun_charge)
 			cranking = TRUE
-			if(do_after(user, 1 SECONDS) && cranking)
+			if(do_after(user, 1 SECONDS) && !fire_interrupted)
```

Comment:
```suggestion
			if(do_after(user, 1 SECONDS) && cranking)
```

You could just double check for cranking again after the do_after and not need a separate variable at all. Set cranking to false when firing instead of a new variable to true.

Reply:
> this is exactly how it functions currently.

It is, but the reason it fails is because there was no check for the variable before calling `crank_charge()` like there is now.